### PR TITLE
Use rules_shell instead of built-in rules

### LINKED ISSUE #3074: Explicitly load rules_shell
When using Bazel 8.0.0 with rules_rust 0.52.2 and `common --incompatible_autoload_externally="+@rules_java"` in `.bazelrc`
I'm getting

```
ERROR: /home/runner/.cache/bazel/_bazel_runner/9f4ed60bba64c867c2242caf607221b5/external/rules_rust++rust+rust_linux_x86_64__x86_64-unknown-linux-gnu__stable_tools/BUILD.bazel:50:1: name 'sh_binary' is not defined (did you mean 'cc_binary'?)
ERROR: /home/runner/.cache/bazel/_bazel_runner/9f4ed60bba64c867c2242caf607221b5/external/rules_rust++rust+rust_linux_x86_64__x86_64-unknown-linux-gnu__stable_tools/BUILD.bazel: no such target '@@rules_rust++rust+rust_linux_x86_64__x86_64-unknown-linux-gnu__stable_tools//:rust_toolchain': target 'rust_toolchain' not declared in package '' defined by /home/runner/.cache/bazel/_bazel_runner/9f4ed60bba64c867c2242caf607221b5/external/rules_rust++rust+rust_linux_x86_64__x86_64-unknown-linux-gnu__stable_tools/BUILD.bazel
ERROR: /home/runner/.cache/bazel/_bazel_runner/9f4ed60bba64c867c2242caf607221b5/external/rules_rust+/util/process_wrapper/BUILD.bazel:31:36: Target '@@rules_rust+//util/process_wrapper:process_wrapper' depends on toolchain '@@rules_rust++rust+rust_linux_x86_64__x86_64-unknown-linux-gnu__stable_tools//:rust_toolchain', which cannot be found: no such target '@@rules_rust++rust+rust_linux_x86_64__x86_64-unknown-linux-gnu__stable_tools//:rust_toolchain': target 'rust_toolchain' not declared in package '' defined by /home/runner/.cache/bazel/_bazel_runner/9f4ed60bba64c867c2242caf607221b5/external/rules_rust++rust+rust_linux_x86_64__x86_64-unknown-linux-gnu__stable_tools/BUILD.bazel'
```

I'll upgrade rules_rust to latest as soon as the BCR woes are fixed, but I've not seen an issue or PR about rules_shell, so I assume this is still current.

### PR DESCRIPTION
**Pull request by jwnimmer-tri**
Closes #3074.

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -29,6 +29,10 @@ bazel_dep(
     name = "rules_license",
     version = "1.0.0",
 )
+bazel_dep(
```
Comment (UebelAndre) [inline MODULE.bazel:32]: Is there no need to add anything to the WORKSPACE file?

**Review by UebelAndre — CHANGES_REQUESTED** [review]
Thanks! Just one question.

File: MODULE.bazel (hunk 1 above)
Comment (jwnimmer-tri) [reply MODULE.bazel:32]: My impression is that anyone using `WORKSPACE` should have already been calling `rules_rust_dependencies()` from it (and indeed, our neighboring `WORKSPACE.bazel` does that) so the change in this PR to `rust/repositories.bzl` to add `rules_shell` to the list should be sufficient.

Comment (UebelAndre) [reply MODULE.bazel:32]: Ah! I missed that, yeah, then it's covered. Thanks!

**Review by UebelAndre — APPROVED** [review]

**Pull request merged**